src/main/java/org/spongepowered/common/entity/EntityUtil.java

```diff
@@ -321,6 +321,7 @@ public static boolean isEntityDead(net.minecraft.entity.Entity entity) {
 
         MoveEntityEvent.Teleport event = SpongeEventFactory.createMoveEntityEventTeleport(Sponge.getCauseStackManager().getCurrentCause(), fromTransform, toTransform, (org.spongepowered.api.entity.Entity) entityIn);
         SpongeImpl.postEvent(event);
+        Sponge.getCauseStackManager().popCause();
```

Comment:
I'm not sure that popping AND using a cause stack frame is appropriate?

Basically, using a cause stack frame in a try catch is using the https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html to push and pop a Cause Stack Frame (similar to the Call Stack Frame)  https://stackoverflow.com/questions/10057443/explain-the-concept-of-a-stack-frame-in-a-nutshell

Sorry if you already understand and I've made a basic error due to looking at a subset of your changes.

Also we much prefer the try style compared to manual push / pop's due to the possibility for error or cause stack corruption.